src/command/hoard_command.rs

```diff
@@ -110,14 +113,26 @@ impl HoardCommand {
         }
     }
 
-    pub fn with_name_input(self, default_value: Option<String>) -> Self {
+    fn with_name_input_prompt(
+        self,
+        default_value: Option<String>,
+        trove: &CommandTrove,
+        prompt_string: String,
+    ) -> Self {
+        let mut command_names = trove
+            .commands
+            .iter()
+            .filter(|x| x.namespace == self.namespace);
+
         let name: String = Input::with_theme(&ColorfulTheme::default())
             .default(default_value.unwrap_or(String::from("")))
-            .with_prompt("Name your command")
+            .with_prompt(prompt_string)
             .validate_with({
                 move |input: &String| -> Result<(), &str> {
                     if input.contains(' ') {
                         Err("The name cant contain whitespaces")
+                    } else if command_names.any(|x| x.name == input.to_string()) {
```

Comment:
If a user supplies a colliding name twice, it gets accepted. 
I _think_ this is due to `command_names` being an iterator. Debugging this line shows that `command_names` in the first iteration consists of all `HoardCommand`. The second iteration only consists of the `HoardCommand` that fulfills `command_names.any(|x| x.name == input.to_string()`. In the third iteration its empty. 
I don't fully understand what's happening, but moving the `iter()` and `filter()` calls into the closure fixes the issue.
  ```
let mut command_names = trove
            .commands
            .iter()
            .filter(|x| x.namespace == self.namespace);
```

This definitely creates more unnecessary clones of all commands. But I think this would be fine for now
```
let mut command_names = trove .commands.clone()
            ...
            ...
} else if command_names.iter().filter(|x| x.namespace == namespace).any(|x| x.name == input.to_string()) {
```